src/vmm/src/persist.rs

```diff
@@ -376,38 +364,32 @@ pub enum ValidateCpuVendorError {
 /// - Failed to read host vendor.
 /// - Failed to read snapshot vendor.
 #[cfg(target_arch = "x86_64")]
-pub fn validate_cpu_vendor(microvm_state: &MicrovmState) -> Result<bool, ValidateCpuVendorError> {
-    let host_vendor_id = get_vendor_id_from_host()?;
-
-    let snapshot_vendor_id = microvm_state.vcpu_states[0]
-        .cpuid
-        .vendor_id()
-        .ok_or(ValidateCpuVendorError::Snapshot)?;
-
-    if host_vendor_id == snapshot_vendor_id {
-        info!("Snapshot CPU vendor id: {:?}", &snapshot_vendor_id);
-        Ok(true)
-    } else {
-        error!(
-            "Host CPU vendor id: {:?} differs from the snapshotted one: {:?}",
-            &host_vendor_id, &snapshot_vendor_id
-        );
-        Ok(false)
+pub fn validate_cpu_vendor(microvm_state: &MicrovmState) {
+    let host_vendor_id = get_vendor_id_from_host();
+    let snapshot_vendor_id = microvm_state.vcpu_states[0].cpuid.vendor_id();
+    match (host_vendor_id, snapshot_vendor_id) {
+        (Ok(host_id), Some(snapshot_id)) => {
+            info!("Host CPU vendor id: {host_id:?}");
+            info!("Snapshot CPU vendor id: {snapshot_id:?}");
+            if host_id != snapshot_id {
+                warn!("Host CPU vendor id differs from the snapshotted one",);
+            }
+        }
+        (Ok(host_id), None) => {
+            info!("Host CPU vendor id: {host_id:?}");
+            warn!("Snapshot CPU vendor id: couldn't get from a snapshot");
```

Comment:
nit: here and below: `s/a snapshot/the snapshot/`

Reply:
done